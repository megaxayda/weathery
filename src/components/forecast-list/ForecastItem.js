import React, { memo } from 'react';
import Col from 'react-bootstrap/Col';
import Card from 'react-bootstrap/Card';
import PropTypes from 'prop-types';

import './forecastItem.css';

const days = ['Sunday', 'Monday', 'Tuesday', 'Wednesday', 'Thursday', 'Friday', 'Saturday'];

function ForecastItem({ date, min, max }) {
  return (
    <Card className="custom_card">
      <Card.Body>
        <Card.Title>{days[new Date(date).getDay()]}</Card.Title>
        <Card.Text>Min: {min}</Card.Text>
        <Card.Text>Max: {max}</Card.Text>
      </Card.Body>
    </Card>
  );
}

export default memo(ForecastItem);

ForecastItem.propTypes = {
  date: PropTypes.string.isRequired,
  min: PropTypes.number.isRequired,
  max: PropTypes.number.isRequired,
};
