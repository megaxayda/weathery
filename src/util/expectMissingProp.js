import sinon from 'sinon';

function expectMissingProp(prop, component) {
  sinon.assert.calledWithMatch(
    console.error,
    new RegExp(`Warning: Failed prop type: The prop \`${prop}\` is marked as required in \`${component}\`, but its value is \`undefined\`.`)
  );
}

export default expectMissingProp;
