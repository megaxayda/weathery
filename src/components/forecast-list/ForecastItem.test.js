import React from 'react';
import { render, unmountComponentAtNode } from 'react-dom';
import { act } from 'react-dom/test-utils';
import PropTypes from 'prop-types';
import sinon from 'sinon';
import renderer from 'react-test-renderer';

import expectMissingProp from 'util/expectMissingProp';
import ForecastItem from './ForecastItem';

let container = null;
beforeEach(() => {
  // setup a DOM element as a render target
  container = document.createElement('div');
  document.body.appendChild(container);
  sinon.stub(console, 'error');
});

afterEach(() => {
  // cleanup on exiting
  unmountComponentAtNode(container);
  container.remove();
  container = null;
  console.error.restore();
  PropTypes.resetWarningCache();
});

test('check props type', () => {
  act(() => {
    render(<ForecastItem />, container);
  });

  sinon.assert.callCount(console.error, 3);
  expectMissingProp('date', 'ForecastItem');
  expectMissingProp('min', 'ForecastItem');
  expectMissingProp('max', 'ForecastItem');
});

test('render forecast card', () => {
  const tree = renderer.create(<ForecastItem date={'2020-05-22'} min={12} max={30} />).toJSON();
  expect(tree).toMatchSnapshot();
});
