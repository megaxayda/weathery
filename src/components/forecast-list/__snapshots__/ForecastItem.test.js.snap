// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render forecast card 1`] = `
<div
  className="custom_card card"
>
  <div
    className="card-body"
  >
    <div
      className="card-title h5"
    >
      Friday
    </div>
    <p
      className="card-text"
    >
      Min: 
      12
    </p>
    <p
      className="card-text"
    >
      Max: 
      30
    </p>
  </div>
</div>
`;
